src/System.Text.JsonLab/System/Text/Json/JsonReaderState.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Text.JsonLab
+{
+    public struct JsonReaderState : IEquatable<JsonReaderState>
```

Comment:
I compare the JsonReaderState to the default value to infer if the caller passed in a state (retry attempt). This type doesn't necessarily need to be IEquatable. The equality operations might not be necessary (if I can find a way to remove `_isRetry`.